src/slider/styled-components.js

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import {styled} from '../styles/index';
+import {} from './constants';
+import {getToggleCheckMarkStyles} from '../checkbox/styled-components';
+
+export const Root = styled('div', props => {
+  return {
+    position: 'relative',
+  };
+});
+
+export const Axis = styled('div', props => {
+  const {$theme} = props;
+  const {} = $theme;
+  let background = 'linear-gradient(to right';
+  const fillColor = '#1B6DE0';
+  const emptyColor = '#CCCCCC';
+  const values = props.$value;
+  const isRange = values.length % 2 === 0;
+  for (let i = 0; i < values.length; i++) {
+    background += ', ';
+    if (isRange) {
+      background += i % 2 === 0 ? emptyColor : fillColor;
+      background += ' ' + values[i] + '%, ';
+      background += i % 2 === 0 ? fillColor : emptyColor;
+      background += ' 0% ';
+    } else {
+      background += i % 2 === 0 ? fillColor : emptyColor;
+      background += ' ' + values[i] + '%, ';
+      background += i % 2 === 0 ? emptyColor : fillColor;
+      background += ' 0% ';
+    }
+  }
+  background += ', ' + emptyColor + ' 0%';
+  background += ')';
```

Comment:
Seems like there are cases where the blue fill can be past the handle?
![image](https://user-images.githubusercontent.com/875591/46684735-920bef00-cba8-11e8-9d52-ca314671a2f2.png)

Reply:
It also feels like using two elements here instead of a linear gradient would simplify a lot of this logic. Also makes it easier for people to customize the fill and background color.

Reply:
fixed this. Also applied approach used in progress bar with couple divs for that